### PR TITLE
COMMERCE-12431 OrgChart View/Edit Panel - Account

### PR DESCRIPTION
**Pull request by smottal**
https://liferay.atlassian.net/browse/COMMERCE-12431

### CODE REVIEW

**Comment by smottal** [comment]
ci:test:sf

**Review by andrea-ale-sbarra — COMMENTED** [review]
Overall looks good.
Make sure the PUT behaviour should change or not.
Thank you!

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 1]
```diff
@@ -7138,6 +7138,8 @@ feature.flag.COMMERCE-11028.description=Allows clients to request a quote for in
 feature.flag.COMMERCE-11028.title=Price on Application
 feature.flag.COMMERCE-11287.description=Sell and manage SKUs by adding different units of measure to the SKU. Configure the unit of measure to optionally allow for sales in decimal quantities. Units of measure have their own pricing and inventory. Once configured, they add to the shopping experience of the buyer who can now purchase SKUs in different units of measure.
 feature.flag.COMMERCE-11287.title=Units of Measure
+feature.flag.COMMERCE-12192.description=This setting enables new features to the management organization chart widget such as searching and viewing/editing information details.
```
[the remark this answers is not in the document]
Comment (smottal) [reply modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:7141]: I had a discussion with @iovaneale about this two weeks ago for the other PR

**Comment by smottal** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:sf

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/AccountResourceImpl.java [hunk 2]
```diff
@@ -311,45 +315,26 @@ public Account putAccount(Long accountId, Account account)
...
 
+		AccountEntry accountEntry =
+			_accountEntryService.fetchAccountEntryByExternalReferenceCode(
+				contextCompany.getCompanyId(), externalReferenceCode);
+
+		if (accountEntry == null) {
+			throw new NoSuchEntryException(
+				"Unable to find Account with external reference code: " +
+					externalReferenceCode);
+		}
```
[the remark this answers is not in the document]
Comment (smottal) [reply modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/AccountResourceImpl.java]: you are right, I was fooled by the method `putAccount(long accountId)` which only does add. I also updated that to manage add or update based on the account existence

File: modules/dxp/apps/commerce/commerce-organization-web/src/main/resources/META-INF/resources/js/utils/LogoSelector.js [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+			/>
+
+			{fileEntryId ? (
+				<img
+					alt={sub(Liferay.Language.get('current-x'), label)}
+					className="logo-selector-img mb-3"
+					src={logoURL}
+				/>
+			) : (
+				<svg className="logo-selector-default-img mb-3">
```
Comment (andrea-ale-sbarra) [inline modules/dxp/apps/commerce/commerce-organization-web/src/main/resources/META-INF/resources/js/utils/LogoSelector.js:112]: ask confirmation to Censi

**Review by andrea-ale-sbarra — COMMENTED** [review]
Overall looks good. Just few comments.
Thank you!

**Comment by smottal** [comment]
ci:test:sf

File: modules/dxp/apps/commerce/commerce-organization-web/src/main/resources/META-INF/resources/js/utils/LogoSelector.js (hunk 3 above)
Comment (smottal) [reply modules/dxp/apps/commerce/commerce-organization-web/src/main/resources/META-INF/resources/js/utils/LogoSelector.js:112]: Updated the account icon to the briefcase (also in the graph)

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/AccountResourceImpl.java [hunk 4]
```diff
@@ -308,48 +268,84 @@ public void postOrganizationAccountsByExternalReferenceCode(
 	public Account putAccount(Long accountId, Account account)
 		throws Exception {
 
+		AccountEntry accountEntry = _accountEntryService.fetchAccountEntry(
+			accountId);
+
+		if (accountEntry == null) {
+			return _toAccount(_addAccountEntry(account));
+		}
```
[the remark this answers is not in the document]
Comment (smottal) [reply modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/AccountResourceImpl.java]: Updated the PR to manage the patch method in a different way

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties (hunk 1 above)
[the remark this answers is not in the document]
Comment (smottal) [reply modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties]: I updated the description as discussed with the PM

**Review by andrea-ale-sbarra — APPROVED** [review]
LGTM.
Thank you!

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant\

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:commerce-functional

**Comment by baharturk** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:sf

**Comment by baharturk** [comment]
ci:close

**Pull request closed without merging**